commcare-cloud-bootstrap/commcare_cloud_bootstrap.py

```diff
@@ -156,31 +262,35 @@ def poll_for_aws_state(env, instance_ids):
     }
     if not unfinished_instances:
         return {
-            instance['InstanceId']: instance['PublicIpAddress']
+            instance['InstanceId']: (instance['PublicIpAddress'], instance['PrivateIpAddress'])
             for instance in instances
             if instance['InstanceId'] in instance_ids
         }
 
 
 def save_inventory(env, inventory):
-    inventory_file_contents = yaml.safe_dump(inventory.to_json(), default_flow_style=False)
+    j2 = jinja2.Environment(loader=jinja2.FileSystemLoader(os.path.dirname(__file__)))
+    template = j2.get_template('inventory.ini.j2')
+    inventory_file_contents = template.render(inventory=inventory)
```

Comment:
Ansible does support YAML inventories - just wondering if you considered that option.

Reply:
Yeah I did consider that, in fact the first versions of this _were_ written in the YAML format. I found it kind of bad to work with, and then there were some ways in which ansible wasn't reading it correctly, possibly because of the `.ini` ending it assumes. Anyway, while that was certainly fixable, I decided to switch to the ini-like syntax. The code for it generating the yaml format was getting worse and worse as I added and changed things, so I opted to refactor the code to work in terms of `Group`s and `Host`s as they're now defined, and then mapping them to and from the file format at the last minute. It's true that even with this refactor, I could be mapping to and from the YAML format, but I like that it's very obvious what's going on when you look at the `inventory.ini.j2` template, rather than having to carefully read the code that assembles it. Of course, that does trade of with the custom parsing code above; but my thought is that it's clear what that function intends to do, even if you don't read every line.

Not sure if these are convincing reasons, but I did like what I ended up with a lot better than what I had before. If the custom parsing or anything else becomes a problem, I'm happy to switch this back to YAML in the future.